src/cephadm/node-proxy/redfish_system.py

```diff
@@ -80,12 +94,27 @@ def stop_update_loop(self):
     def update(self):
         #  this loop can have:
         #  - caching logic
-        while self.run:
-            self._update_system()
-            # following calls in theory can be done in parallel
-            self._update_metadata()
-            self._update_memory()
-            self._update_power()
-            self._update_network()
-            self._update_storage()
-            sleep(3)
+        try:
+            while self.run:
+                log.debug("waiting for a lock.")
+                self.lock.acquire()
+                log.debug("lock acquired.")
+                try:
+                    self._update_system()
+                    # following calls in theory can be done in parallel
+                    self._update_metadata()
+                    self._update_memory()
+                    self._update_power()
+                    self._update_network()
+                    self._update_processors()
+                    self._update_storage()
+                    self.data_ready = True
+                    sleep(5)
```

Comment:
we have to think about where we will read this module configuration (intervals, etc). It should be good to have some centralized place to get the conf from.